### ISSUE TITLE
Build for Nix 2.8.1 fails

### ISSUE THREAD

**Issue by jakubgs** [shlevy/nix-plugins#13]
Fails with:
```
[ 50%] Building CXX object CMakeFiles/nix-extra-builtins.dir/extra-builtins.cc.o
/build/source/extra-builtins.cc: In function 'void extraBuiltins(nix::EvalState&, const nix::Pos&, nix::Value**, nix::Value&)':
/build/source/extra-builtins.cc:45:19: error: 'class nix::EvalState' has no member named 'mkAttrs'
   45 |             state.mkAttrs(*arg, 2);
      |                   ^~~~~~~
/build/source/extra-builtins.cc:57:9: error: 'mkApp' was not declared in this scope
   57 |         mkApp(v, *fun, *arg);
      |         ^~~~~
/build/source/extra-builtins.cc:58:25: error: no matching function for call to 'nix::EvalState::forceValue(nix::Value&)'
   58 |         state.forceValue(v);
      |         ~~~~~~~~~~~~~~~~^~~
In file included from /nix/store/a9cdlvzqv9z8abnb5ywy12lp81cavh90-nix-2.8.1-dev/include/nix/primops.hh:3,
                 from /build/source/extra-builtins.cc:2:
/nix/store/a9cdlvzqv9z8abnb5ywy12lp81cavh90-nix-2.8.1-dev/include/nix/eval.hh:230:17: note: candidate: 'template<class Callable> void nix::EvalSta>
  230 |     inline void forceValue(Value & v, Callable getPos);
      |                 ^~~~~~~~~~
/nix/store/a9cdlvzqv9z8abnb5ywy12lp81cavh90-nix-2.8.1-dev/include/nix/eval.hh:230:17: note:   template argument deduction/substitution failed:
/build/source/extra-builtins.cc:58:25: note:   candidate expects 2 arguments, 1 provided
   58 |         state.forceValue(v);
      |         ~~~~~~~~~~~~~~~~^~~
In file included from /nix/store/a9cdlvzqv9z8abnb5ywy12lp81cavh90-nix-2.8.1-dev/include/nix/eval.hh:512,
                 from /nix/store/a9cdlvzqv9z8abnb5ywy12lp81cavh90-nix-2.8.1-dev/include/nix/primops.hh:3,
                 from /build/source/extra-builtins.cc:2:
/nix/store/a9cdlvzqv9z8abnb5ywy12lp81cavh90-nix-2.8.1-dev/include/nix/eval-inline.hh:102:6: note: candidate: 'void nix::EvalState::forceValue(nix:>
  102 | void EvalState::forceValue(Value & v, const Pos & pos)
      |      ^~~~~~~~~
/nix/store/a9cdlvzqv9z8abnb5ywy12lp81cavh90-nix-2.8.1-dev/include/nix/eval-inline.hh:102:6: note:   candidate expects 2 arguments, 1 provided
/build/source/extra-builtins.cc:62:9: error: 'mkNull' was not declared in this scope
   62 |         mkNull(v);
      |         ^~~~~~
/build/source/extra-builtins.cc: In function 'void cflags(nix::EvalState&, const nix::Pos&, nix::Value**, nix::Value&)':
/build/source/extra-builtins.cc:72:11: error: 'class nix::EvalState' has no member named 'mkAttrs'
   72 |     state.mkAttrs(v, 3);
      |           ^~~~~~~
make[2]: *** [CMakeFiles/nix-extra-builtins.dir/build.make:76: CMakeFiles/nix-extra-builtins.dir/extra-builtins.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/nix-extra-builtins.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by jakubgs** [comment]
It appears it builds fine when I switch to using `unstable.nixVersions.nix_2_9` with `pkgs.unstable.nix-plugins`, but now I get:
```
error: could not dynamically open plugin file '/nix/store/n0bah9smrfpmh0bkj4fj9nkg3byfpbhq-nix-plugins-9.0.0/lib/nix/plugins/libnix-extra-builtins.so': /nix/store/n0bah9smrfpmh0bkj4fj9nkg3byfpbhq-nix-plugins-9.0.0/lib/nix/plugins/libnix-extra-builtins.so: undefined symbol: _ZN3nix17prim_importNativeERNS_9EvalStateERKNS_3PosEPPNS_5ValueERS5
```
When running `nixos-rebuild`.

**Comment by ibizaman** [comment]
The commit that updates to 9.0.0 in nixpkgs is https://github.com/NixOS/nixpkgs/commit/ad9a297ff4b7a0065c42c13553cc669f0b121f0c but it is not included in [22.05](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/development/libraries/nix-plugins/default.nix).

In my `configuration.nix`, I put:
```nix
let
  nix-plugins = pkgs.nix-plugins.overrideAttrs (oldAttrs: rec {
    version = "9.0.0";
    buildInputs = [ pkgs.nix pkgs.boost ];
    src = pkgs.fetchFromGitHub {
      owner = "shlevy";
      repo = "nix-plugins";
      rev = "9.0.0";
      hash = "sha256-AkHsZpYM4EY8SNuF6LhxF2peOjp69ICGc3kOLkDms64=";
    };
  });
in
{
  nix.extraOptions = ''
    plugin-files = ${nix-plugins}/lib/nix/plugins
  '';
}
```

**Comment by shlevy** [comment]
With `9.0.0` released this is no longer a `nix-plugins` issue. If need be someone can backport 9.0.0 as `nix-plugins_9` in `22.05`.
